### ISSUE TITLE
Transform which ran with v1 not working with v3

### ISSUE THREAD

**Issue by rreusser** [browserify/static-module#48]
Hi! [cwise](https://github.com/scijs/cwise) has had some long-running PRs to upgrade its usage of static-module and get rid of some security warnings, but it seems that the particular usage static-module is no longer functioning under static-module 3.*.

The [cwise transform](https://github.com/scijs/cwise/blob/master/lib/cwise-transform.js) finds references to `var cwise = require('cwise'); cwise({...})` and replaces them with evaluated code.

I'm guessing a bit, but it seems like maybe the usage as a bare function require as opposed to properties on the require (i.e. `require('cwise')(...)` as opposed to `require('fs).readFileSync(...)`) are not working.

FWIW, the transform _does_ still function and get triggered correctly.

I've detailed a test case [here](https://github.com/scijs/cwise/pull/25#issuecomment-442942342).

Glad to debug a bit further, but I'd thought I'd check to see if this might just ### need a small API usage update instead of involved debugging. Thanks!

**Comment by goto-bus-stop** [comment]
i'll have a look at this tomorrow. the big change in v3 was that unknown uses of a module are now kept, instead of removed (risking runtime errors). but v3 also added scope tracking which was a fairly large refactor that may have broken some stuff!

`require('xyz')('')` is at least "intended" to work, but maybe it's bugged.

**Comment by archmoj** [comment]
@goto-bus-stop possibly any update on this?

**Comment by goto-bus-stop** [comment]
no, i forgot about it :smile: i'll aim to work on this on Friday. a PR would be welcome as well of course.  I think the way to go here is to change the current code to pass in ASTs, which can be exposed as `staticModule.ast`, then implement the current API in terms of that.

**Comment by archmoj** [comment]
Many thanks @goto-bus-stop.
That would be great news for maintaining many webgl modules.
cc: @etpinard

**Comment by archmoj** [comment]
@goto-bus-stop My apologies for asking one more time.
Wondering if there may be progress on this issue? 
Many thanks.

**Comment by goto-bus-stop** [comment]
The refactor is a bit trickier than I anticipated, and I haven't had the time to make it work yet unfortunately :(

**Comment by Domino987** [comment]
Hi @goto-bus-stop , i was wondering if you could refactor it?

**Comment by telamonian** [comment]
Hey @goto-bus-stop, thanks for working on fixing the issues with cwise and static-module. I have a professional interest in seeing cwise get upgraded past its security issues. Currently at my company, cwise's security issues are a blocker for all of the (way upstream) Jupyterlab Plotly extensions, which all depend on cwise through numerous dependency chains.

If you're still interested in working on this fix, I'd be more than glad to pitch in some of my time to help. Is your work so far on the refactor publicly viewable anywhere?

**Comment by goto-bus-stop** [comment]
@telamonian I think I tried a few approaches but none of them worked out so I discarded them.

For now @archmoj's approach where we just add an option to opt back into the unsafe behaviour is likely best—it's only insecure if you use it on untrusted code anyway…

**Comment by goto-bus-stop** [comment]
#56 updates to the static-eval with @archmoj's fix but I'm not sure if it's enough. I tried using that branch in my local cwise and passing `{ allowAccessToMethodsOnFunctions: true }` as options but the output of the snippet in https://github.com/scijs/cwise/pull/25#issuecomment-442942342 still contains a full JS parser. could one of you confirm?

**Comment by archmoj** [comment]
> #56 updates to the static-eval with @archmoj's fix but I'm not sure if it's enough. I tried using that branch in my local cwise clone and passing `{ allowAccessToMethodsOnFunctions: true }` as options but the output of the snippet in [scijs/cwise#25 (comment)](https://github.com/scijs/cwise/pull/25#issuecomment-442942342) still contains a full JS parser. could one of you confirm?

@goto-bus-stop thanks for the follow ups. It works. And with cwise transform option there is no unused parser in the bundle.
Please see my new PR at https://github.com/scijs/cwise/pull/32 that make use of your #56 PR.

**Comment by archmoj** [comment]
@goto-bus-stop update: you are right. There is still that parser problem.

**Comment by goto-bus-stop** [comment]
hm, static-eval might be bailing out somewhere else as well then :/

**Comment by archmoj** [comment]
Please note that the static-module/static-eval bump issue was addressed in [plotly.js v1.54.4](https://github.com/plotly/plotly.js/releases/tag/v1.54.4) by not using `cwise`.

**Comment by telamonian** [comment]
> Please note that the static-module/static-eval bump issue was addressed in [plotly.js v1.54.4](https://github.com/plotly/plotly.js/releases/tag/v1.54.4) by not using `cwise`.

Hooray!
